scripts/ScriptUtilities.psm1

```diff
@@ -33,19 +33,89 @@ function Exit-WithFailureMessage($scriptName, $message) {
     exit 1
 }
 
+# NuGet Package Creation section
+function New-NuGetPackageFromProjectFile($Configuration, $project, $version) {
+    $PackageOutputDirectory = "$BinRoot\NuGet\$Configuration"
+    $projectFile = "$SourceRoot\$project\$project.csproj"
+
+    $arguments =
+        "pack", $projectFile,
+        "--configuration", $Configuration,
+        "--no-build", "--no-restore",
+        "--include-source", "--include-symbols",
+        "-p:Platform=$Platform",
+        "--output", $PackageOutputDirectory
+
+    Write-Debug "dotnet $($arguments -join ' ')"
+
+    dotnet $arguments
+    if ($LASTEXITCODE -ne 0) {
+        Exit-WithFailureMessage $ScriptName "$project NuGet package creation failed."
+    }
+}
+
+function New-NuGetPackageFromNuspecFile($Configuration, $project, $version, $suffix = "") {
+    $PackageOutputDirectory = "$BinRoot\NuGet\$Configuration"
```

Comment:
>$PackageOutputDirectory [](start = 4, length = 23)

I don't think it's a good idea for a module to export a variable whose value depends on the last script that was run. Here's what I suggest:

1. In this module, define and export the variable `$PackageOutputDirectoryRoot = "$BinRoot\NuGet"`

2. In this module, define and export the function
```
function Get-PackageDirectoryName($configuration) {
    Join-Path $PackageOutputDirectoryRoot $configuration
}
```

3. Wherever you currently use `$PackageOutputDirectory` (including here), replace it with a call to `Get-PackageDirectoryName $configuration`.